client/components/app-promo/test/index.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import { createStore } from 'redux';
+import { Provider } from 'react-redux';
+import { expect } from 'chai';
+import { mount } from 'enzyme';
+import { spy, stub } from 'sinon';
+import mockery from 'mockery';
+
+import useFakeDom from 'test/helpers/use-fake-dom';
+import useMockery from 'test/helpers/use-mockery';
+
+describe( 'AppPromo', ( ) => {
+	let AppPromo;
+	const store = createStore( stub() );
+
+	useFakeDom();
+	useMockery();
+
+	const desktopPromoKey = 'desktop_promo_dismissed';
+	let desktopPromoDismissed = false;
+	let localeSlug = 'en';
+	let isMobile = false;
+	let isChromeOS = false;
+
+	const storeMock = {
+		get: ( val ) => {
+			if ( val === desktopPromoKey ) {
+				return desktopPromoDismissed;
+			}
+
+			return undefined;
+		},
+		set: spy()
+	};
+
+	const userUtilsMock = { getLocaleSlug: ( ) => localeSlug };
+	const viewportMock = { isMobile: ( ) => isMobile };
+	const userAgentUtilsMock = { isChromeOS: ( ) => isChromeOS };
+	const appPromoDetails = {
+		promo_code: 'a0001',
+		message: 'WordPress.com your way  — desktop app now available for Mac, Windows, and Linux.'
+	};
+	const appPromoLink = 'http://www.wordpress.com';
+	const promoRetrieverMock = {
+		getRandomPromo: ( ) => appPromoDetails,
+		getPromoLink: ( ) => appPromoLink
+	};
+
+	const appStoreComponent = ( ) => {
+		return (
+			<Provider store={ store } >
+				<AppPromo location="reader" />
+			</Provider>
+			);
+	};
+
+	before( ( ) => {
+		mockery.registerMock( 'store', storeMock );
+		mockery.registerMock( 'lib/user/utils', userUtilsMock );
+		mockery.registerMock( 'lib/user-agent-utils', userAgentUtilsMock );
+		mockery.registerMock( 'lib/viewport', viewportMock );
+		mockery.registerMock( './lib/promo-retriever', promoRetrieverMock );
+
+		AppPromo = require( '../' );
+	} );
+
+	beforeEach( ( ) => {
+		desktopPromoDismissed = false;
+		localeSlug = 'en';
+		isMobile = false;
+		isChromeOS = false;
+	} );
+
+	it( 'should render nothing when the promo has already been dismissed', ( ) => {
+		desktopPromoDismissed = true;
+		const wrapper = mount( appStoreComponent() );
+		expect( wrapper.find( '.app-promo' ) ).to.have.lengthOf( 0 );
```

Comment:
did you try shallow rendering here? I'm not sure if it would work or not but if it does, it would be much faster